src/models/ModelBuilder.cc

```diff
@@ -4,24 +4,38 @@
 
 #include "ModelBuilder.h"
 
-void ModelBuilder::processSubstitutions(Map<PTRef,PtAsgn,PTRefHash> const & subst) {
-    Map<PTRef,PtAsgn,PTRefHash> copy;
+void ModelBuilder::processSubstitutions(MapWithKeys<PTRef,PtAsgn,PTRefHash> const & subst) {
+    MapWithKeys<PTRef,PtAsgn,PTRefHash> copy;
     subst.copyTo(copy);
     logic.substitutionsTransitiveClosure(copy);
     auto assignCopy = this->assignment;
     auto model = this->build();
-    auto entries = copy.getKeysAndValsPtrs();
-    for (auto const * const entry : entries) {
-        if (logic.isIte(entry->key)) {
+    for (auto key : copy.getKeys()) {
+        if (logic.isIte(key)) {
             // We store only values of reals variables
             continue;
         }
-        assert(logic.isVar(entry->key));
-        if (entry->data.sgn == l_True) {
-            PTRef mappedTerm = entry->data.tr;
+        assert(logic.isVar(key));
+        PtAsgn target = copy[key];
+        if (target.sgn == l_True) {
+            PTRef mappedTerm = target.tr;
             PTRef val = model->evaluate(mappedTerm);
             assert(logic.isConstant(val));
-            assignCopy.insert(std::make_pair(entry->key, val));
+            assignCopy.insert(std::make_pair(key, val));
+        }
+    }
+    for (auto key : copy.getKeys()) {
+        if (logic.isIte(key)) {
+            // We store only values of reals variables
+            continue;
+        }
+        assert(logic.isVar(key));
+        PtAsgn target = copy[key];
+        if (target.sgn == l_True) {
+            PTRef mappedTerm = target.tr;
+            PTRef val = model->evaluate(mappedTerm);
+            assert(logic.isConstant(val));
+            assignCopy.insert(std::make_pair(key, val));
```

Comment:
Looks like the same code twice. Copy-paste error?

Reply:
Indeed, strange!